src/c_api.rs

```diff
@@ -195,46 +200,40 @@ pub extern "C" fn JSONAPI_getStringFromKey(
         let key = unsafe { &mut *key };
         let path = unsafe { CStr::from_ptr(path).to_str().unwrap() };
         if let Ok(s) = key.redis_json.to_string(path, Format::JSON) {
-            if let Ok(s) = CString::new(s) {
-                unsafe {
-                    *str = s.as_bytes_with_nul().as_ptr() as *const c_char;
-                    *len = s.as_bytes().len();
-                }
-                key.cstr_val = Some(s);
-                return Status::Ok as c_int;
-            }
+            set_string(s.as_str(), str, len);
+            return Status::Ok as c_int;
         }
     }
     Status::Err as c_int
 }
 
 #[no_mangle]
-pub extern "C" fn JSONAPI_getRedisModuleString(
+pub extern "C" fn JSONAPI_getJSON(
     json: JSONApiPathRef,
+    ctx: *mut rawmod::RedisModuleCtx,
     str: *mut *mut rawmod::RedisModuleString,
 ) -> c_int {
     if !json.is_null() {
         let json = unsafe { &*json };
-        if let Some(ref p) = json.path {
-            if let Ok(res) = json.json_key.redis_json.to_string(p, Format::JSON) {
-                return json.json_key.create_rmstring(res.as_str(), str);
-            }
+        if let Ok(res) = serde_json::to_string(json) {
```

Comment:
This, on the other hand, should be serializing (`.to_string(path, Format::JSON`).

Reply:
@rafie Right, we are serializing, but fixed to serialize directly from the current `Value` instead of traversing to it using `path`.